apis/v1alpha2/referencepolicy_types.go

```diff
@@ -95,6 +95,7 @@ type ReferencePolicyFrom struct {
 	// additional resources, the following Route types are part of the "Core"
 	// support level for this field:
 	//
+	// * Gateway
```

Comment:
I think we need some more nuance here. I'm not really sure how this will scale since we really need to document each valid pair, ie HTTPRoute -> Secret doesn't make sense, and there's not currently any way to do Gateway -> Service. I think the best option here is to add a note here that ReferencePolicy is only supported as explicitly documented on object reference fields. Then we can add a comment on the cert ref namespace field in Gateways that references to Secrets in other namespaces can be allowed by ReferenceGrant. Similarly, we can update the BackendRef namespace field to state that references to Services in other namespaces can be allowed by ReferenceGrant.

Reply:
The status Condition -> Reason mappings (see example in https://github.com/kubernetes-sigs/gateway-api/pull/1114) may provide a pattern for clarifying these relationships a bit.